packages/server/src/adapters/aws-lambda.ts

```diff
@@ -0,0 +1,80 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { URLSearchParams } from 'url';
+import { resolveHTTPResponse } from '..';
+import { HTTPHeaders, HTTPRequest } from '../http/internals/types';
+import { AnyRouter, inferRouterContext } from '../router';
+import type { AWSLambdaOptions, LambdaContext } from './lambda-utils';
+
+export type { CreateLambdaContextOptions } from './lambda-utils';
+
+function lambdaEventToHTTPRequest(event: APIGatewayProxyEvent): HTTPRequest {
+  const query = new URLSearchParams();
+  for (const [key, value] of Object.entries(
+    event.multiValueQueryStringParameters ?? {},
+  )) {
+    for (const v of value ?? []) {
+      query.append(key, v);
+    }
+  }
+  return {
+    method: event.httpMethod,
+    query,
+    headers: event.headers,
+    body: event.body,
+  };
+}
+
+function transformHeaders(
+  headers: HTTPHeaders,
+): APIGatewayProxyResult['headers'] {
+  const obj: APIGatewayProxyResult['headers'] = {};
+
+  for (const [key, value] of Object.entries(headers)) {
+    if (typeof value === 'undefined') {
+      continue;
+    }
+    obj[key] = Array.isArray(value) ? value.join(', ') : value;
+  }
+  return obj;
+}
+
+export function createApiGatewayHandler<TRouter extends AnyRouter>(
+  opts: AWSLambdaOptions<TRouter>,
+): (
+  event: APIGatewayProxyEvent,
+  context: LambdaContext,
+) => Promise<APIGatewayProxyResult> {
+  return async (event: APIGatewayProxyEvent, context: LambdaContext) => {
```

Comment:
```suggestion
  return async (event, context) => {
```
inferred from return annotation